src/core/a-assets.js

```diff
@@ -10,126 +10,127 @@ var warn = debug('core:a-assets:warn');
 /**
  * Asset management system. Handles blocking on asset loading.
  */
-module.exports = registerElement('a-assets', {
-  prototype: Object.create(ANode.prototype, {
-    createdCallback: {
-      value: function () {
-        this.isAssets = true;
-        this.fileLoader = fileLoader;
-        this.timeout = null;
-      }
-    },
-
-    attachedCallback: {
-      value: function () {
-        var self = this;
-        var i;
-        var loaded = [];
-        var mediaEl;
-        var mediaEls;
-        var imgEl;
-        var imgEls;
-        var timeout;
-
-        if (!this.parentNode.isScene) {
-          throw new Error('<a-assets> must be a child of a <a-scene>.');
-        }
-
-        // Wait for <img>s.
-        imgEls = this.querySelectorAll('img');
-        for (i = 0; i < imgEls.length; i++) {
-          imgEl = fixUpMediaElement(imgEls[i]);
-          loaded.push(new Promise(function (resolve, reject) {
-            // Set in cache because we won't be needing to call three.js loader if we have.
-            // a loaded media element.
-            THREE.Cache.files[imgEls[i].getAttribute('src')] = imgEl;
-            imgEl.onload = resolve;
-            imgEl.onerror = reject;
-          }));
-        }
+class AAssets extends ANode {
+  constructor (self) {
+    super(self);
```

Comment:
as explained in here https://github.com/aframevr/aframe/pull/4167/files/a3041384c73e829bfe91b110afa7e8a6bf30af5f#r285005172 you need to upgrade self to either itself or the returned node.

I also suggest to cut the mustard with a base class that does just this:
https://github.com/WebReflection/document-register-element#skipping-the-caveat-through-extends

so that you'll have a proper `init() {}` method that won't have ever issues and there's less to explain to newcomers.